test cases/nasm/2 asm language/meson.build

```diff
@@ -0,0 +1,26 @@
+project('test', 'c')
+
+nasm = find_program('nasm', required: false)
```

Comment:
Does this check work with a binary subproject providing nasm like we do in gstreamer?

https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/main/subprojects/win-nasm/meson.build#L24

Reply:
That check if for the unit test only. I don't think meson actually look for compilers using overridden programs, so it probably won't work for gstreamer indeed. The path needs to come from a machine file, or from NASM=/path/nasm.

Reply:
@nirbheek why does GStreamer install nasm like that instead of requiring user to install it using choco for example? To make things easier, I added a commit in this PR to lookup nasm into its default location because it is not in PATH by default on Windows.

Reply:
This is fixed with https://github.com/mesonbuild/meson/pull/10938, but I don't think it should block this PR.